### DISCUSSION TITLE
Is there a way to trigger a callback function when validation fails and pass the errors object to the function?

### DISCUSSION THREAD

**Discussion by arthu** [react-hook-form/react-hook-form#2152] [Q&A] [answered]
I would like to implement some workaround with the callback function
- focus on the first field of errors by the order defined on the screen, not the registered order (due to a few dynamic fields added by the questionnaire, ex. when a user answer "Yes", there are more questions inserted underneath)
- pass analytics some data for product people to know what are the common errors made by users. 

Thanks a lot for your help!!

**Comment by bluebill1049** [comment, marked as answer]
is this what you after? https://github.com/react-hook-form/react-hook-form/issues/1574

**Comment by arthu** [comment]
Thanks @bluebill1049! Yes, it looks like the feature I was looking for. The issue has been closed, has it not been implemented?

**Reply by bluebill1049** [reply]
it's pending for upvote.  https://github.com/react-hook-form/react-hook-form/issues/1981
